data-transform.qmd

```diff
@@ -473,76 +467,73 @@ For example, the following code finds the most delayed flight to each destinatio
 
 ```{r}
 flights |> 
-  group_by(dest) |> 
-  slice_max(arr_delay, n = 1)
+  slice_max(arr_delay, n = 1, by = dest)
 ```
 
 This is similar to computing the max delay with `summarize()`, but you get the whole row instead of the single summary:
 
 ```{r}
 flights |> 
-  group_by(dest) |> 
-  summarize(max_delay = max(arr_delay, na.rm = TRUE))
+  summarize(
+    .by = dest,
+    max_delay = max(arr_delay, na.rm = TRUE),
+  )
 ```
 
-### Grouping by multiple variables
-
-You can create groups using more than one variable.
-For example, we could make a group for each day:
+You can also use `.by` with `filter()` to express the same idea in a slightly different way:
 
 ```{r}
-daily <- flights |>  
-  group_by(year, month, day)
-daily
+flights |> 
+  filter(
+    .by = dest,
+    arr_delay == max(arr_delay, na.rm = TRUE),
+  )
 ```
 
-When you summarize a tibble grouped by more than one variable, each summary peels off the last group.
-In hindsight, this wasn't great way to make this function work, but it's difficult to change without breaking existing code.
-To make it obvious what's happening, dplyr displays a message that tells you how you can change this behavior:
+But as you can see, `slice_max()` takes care of a lot of the details for you.
+
+### Grouping by multiple variables
+
+You can group by multiple variables with `c()`:
 
 ```{r}
-daily_flights <- daily |> 
+flights |> 
   summarize(
-    n = n()
+    .by = c(year, month, day),
+    n = n(),
   )
 ```
 
-If you're happy with this behavior, you can explicitly request it in order to suppress the message:
+`.by` uses the same specification as `select()`, so you could also use `:`, `starts_with()`, `ends_with()`, or any other helper to select the variables:
 
 ```{r}
-#| results: false
-
-daily_flights <- daily |> 
+flights |> 
   summarize(
-    n = n(), 
-    .groups = "drop_last"
+    .by = year:day,
+    n = n(),
   )
 ```
 
-Alternatively, change the default behavior by setting a different value, e.g. `"drop"` to drop all grouping or `"keep"` to preserve the same groups.
-
-### Ungrouping
+### Grouped `mutate()`
 
-You might also want to remove grouping outside of `summarize()`.
-You can do this with `ungroup()`.
+It's also possible to use `.by` with `mutate()`.
+This is particularly useful with the functions that you'll learn about in @sec-numbers, which allow you to do things like rank observations within a group:
 
 ```{r}
-daily |> 
-  ungroup() |>
-  summarize(
-    delay = mean(dep_delay, na.rm = TRUE), 
-    flights = n()
+flights |> 
+  mutate(
+    .by = year:day,
+    rank = min_rank(dep_delay),
+    .after = dep_delay,
```

Comment:
particularly odd to have `.by` separated from `.after` - both optional args

Reply:
+1